refactor: Mission APIWorker, Repository 폴더 구조 리펙토링 작업

Do-hyun-Kim opened the pull request:
## 작업 내용 🧑‍💻
- Domain Layer `Entities` MissionContentResponse, TodayMissionResponse 네이밍 수정
- GetTodayMissionUseCase 파일 제거


## 테스트 케이스 ✅

- [ ] MissionContentResponse, TodayMissionResponse 네이밍 수정하여 `Cannot find type MissionContentResponse` or `Cannot find type TodayMissionResponse` 오류 있는지 확인하기

Do-hyun-Kim committed:
refactor: Domain Entities 네이밍 MissionContentResponse, TodayMissionResponse로 수정
- GetTodayMissionUseCase 파일 제거

akrudal commented on 14th-team5-iOS/Domain/Sources/Mission/Entities/MissionContentResponse.swift:
```diff
@@ -1,20 +1,18 @@
 //
-//  MissionContentData.swift
```
저희.. domain 레이어도 리팩토링하기로 결정되었나욥..?
제가 생각했을 때 도메인 레이어는 DATA로 표기하는게 더 알맞다고 생각하거든욥..

response는 비즈니스적 입장인 엔티티보다 통신에 관련한 엔티티라고 생각이 들어서요...
언제 도메인 레이어의 엔티티가 response 네이밍으로 결정되었나욥..

rlarjsdn3 replied:
지난 오프라인 회의 때 뒤에 DTO 빼고 Entity 이름 붙이기로 했어요

Do-hyun-Kim replied:
아하 그럼 이부분은 `MissionContentEntity`로 수정하는 방향으로 갈까요??

akrudal replied:
@rlarjsdn3
아하 제가 도메인은 나중에하기로 해서 까먹었나보네요. 
노션에도 다시 볼 수 없는데, 혹시 도메인레이어는 나중에 리팩토링하기로 했으니 다시 얘기해볼 수 있을까요?

akrudal replied:
아 오프라인...이여..?

Do-hyun-Kim replied:
온라인 회의라고 말하시는 것 같네용 😄

akrudal replied:
Entity라고 붙이는 것도 좋은 방법 같아여!

akrudal commented on 14th-team5-iOS/Data/Sources/Mission/DataMapping/MissionContentDTO.swift:
```diff
@@ -23,7 +23,7 @@ struct MissionContentDTO: Decodable {
 
 extension MissionContentDTO {
-    func toDomain() -> MissionContentData {
+    func toDomain() -> MissionContentResponse {
```
그..위에를 바꿔야하지않을까여.. MissionContentResponseDTO로여..ㅎㅎ

Do-hyun-Kim replied:
네넵!! `MissionContentDTO`를 `MissionContentResponseDTO`로 수정하겠습니당 👍 👍

akrudal commented on 14th-team5-iOS/Data/Sources/Mission/DataMapping/TodayMissionDTO.swift:
```diff
@@ -14,7 +14,7 @@ struct GetTodayMissionResponse: Codable {
     let content: String
     
-    func toDomain() -> TodayMissionData {
+    func toDomain() -> TodayMissionResponse {
```
파일명도 TodayMissionResponseDTO이렇게 맞추기로 하지않았나용..?

Do-hyun-Kim replied:
아 이부분도 제가 놓쳤네요 `TodayMissionResponseDTO`로 수정하겠습니다 👍 👍

akrudal reviewed (changes requested):
데이터레이어 네이밍 규칙 다시 맞춰주셔야할 것 같습니당

Do-hyun-Kim committed:
fix: MissionContentResponseDTO, GetTodayMissionResponseDTO로 파일 네이밍 수정
- MissionAPIWorker MissionContentResponseDTO, GetTodayMissionResponseDTO 반환 값 변경

akrudal reviewed (approved):
고생하셨습니다 :)
